lib/run/summary.js

```diff
@@ -12,8 +12,20 @@ var _ = require('lodash'),
  */
 RunSummary = function RunSummary (emitter, options) {
     // keep a copy of this instance since, we need to refer to this from various events
-    var summary = this;
-
+    var summary = this,
+        onlyCLIReporterEnabled,
+        skipExecutionsSummary;
+
+    // Need to do add this check for unit-tests when no options are passed
+    if (options) {
+        if (!options.skipExecutionsSummary) {
+            onlyCLIReporterEnabled = (Array.isArray(options.reporters) &&
+                options.reporters.length === 1 &&
+                options.reporters.includes('cli'));
+        }
+        // skipExecutionsSummary in case the option is specified, or only cli is specified as reporter
+        skipExecutionsSummary = options.skipExecutionsSummary || onlyCLIReporterEnabled;
+    }
```

Comment:
Multiple levels of nested `if` blocks seems a little yukk 😬

One way to remove this can be to update the `options.skipExecutionsSummary` itself when it is not set by the user and it is implicit, like so:

```suggestion
    if (!options.skipExecutionSummary) {
    	// skipExecutionsSummary in case the option is specified, or only cli is specified as reporter
    	options.skipExecutionSummary = (Array.isArray(options.reporters) &&
        	options.reporters.length === 1 &&
            options.reporters.includes('cli'));
    }
```

This is just one way, I'm open to more creative/better solutions. 🙏